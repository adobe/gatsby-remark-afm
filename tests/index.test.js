/*
Copyright 2020 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/
const doNotLocalize = require('../src/DoNotLocalize');
const admonitions = require('../src/Admonitions');
const includeRelative = require('../src/IncludeRelative');
// const tabbedCodeBlocks = require("../src/TabbedCodeBlocks");
const { getMarkdownASTForFile, parseASTToMarkdown } = require('./helpers/markdown');
const plugin = require('../index');

const { getAstData, testTrees } = require('./helpers/astData');

describe('includeRelative', () => {
  it('is truthy', () => {
    expect(includeRelative).toBeTruthy();
  });
  it('can detect and render include within markdown which itself has an include', async () => {
    const markdownAST = getMarkdownASTForFile('include-relative', true);
    const processedAST = await plugin({ markdownAST });
    const includeWithInclude = getAstData(testTrees.includeWithInclude);

    expect(processedAST).toEqual(includeWithInclude);
  });

  it('can properly render admonitions within the contents of the included files', async () => {
    const markdownAST = getMarkdownASTForFile('aepcs-api-reference', true);
    const processedAST = await plugin({ markdownAST });
    const includesWithAdmonitions = getAstData(testTrees.includesWithAdmonitions);

    expect(processedAST).toEqual(includesWithAdmonitions);
  });
});

describe('doNotLocalize', () => {
  it('is truthy', () => {
    expect(doNotLocalize).toBeTruthy();
  });
  it('can transform !DNL tags', async () => {
    const markdownAST = getMarkdownASTForFile('do-not-localize', true);
    const processedAST = await plugin({ markdownAST });
    expect(parseASTToMarkdown(processedAST)).toMatchInlineSnapshot(`
      "# Don't Delete:

      For some reason we need at least one markdown file in the markdown-pages directory in order for the build to succeed

      Adobe
      "
    `);
  });
});

describe('admonitions', () => {
  it('is truthy', () => {
    expect(admonitions).toBeTruthy();
  });
  it('can transform !Note tags', async () => {
    const markdownAST = getMarkdownASTForFile('admonitions-note', true);
    const processedAST = await plugin({ markdownAST });
    expect(parseASTToMarkdown(processedAST)).toMatchInlineSnapshot(`
      "# Don't Delete:

      For some reason we need at least one markdown file in the markdown-pages directory in order for the build to succeed

      <Alert header=\\"Note\\" variant=\\"info\\">This is a standard NOTE block.</Alert>
      "
    `);
  });
  it('can transform !Caution tags', async () => {
    const markdownAST = getMarkdownASTForFile('admonitions-caution', true);
    const processedAST = await plugin({ markdownAST });
    expect(parseASTToMarkdown(processedAST)).toMatchInlineSnapshot(`
      "# Don't Delete:

      For some reason we need at least one markdown file in the markdown-pages directory in order for the build to succeed

      <Alert header=\\"Caution\\" variant=\\"error\\">This is a standard CAUTION block.</Alert>
      "
    `);
  });
  it('can transform !Warning tags', async () => {
    const markdownAST = getMarkdownASTForFile('admonitions-warning', true);
    const processedAST = await plugin({ markdownAST });
    expect(parseASTToMarkdown(processedAST)).toMatchInlineSnapshot(`
      "# Don't Delete:

      For some reason we need at least one markdown file in the markdown-pages directory in order for the build to succeed

      <Alert header=\\"Warning\\" variant=\\"warning\\">This is a standard WARNING block.</Alert>
      "
    `);
  });
  it('can transform !Tip tags', async () => {
    const markdownAST = getMarkdownASTForFile('admonitions-tip', true);
    const processedAST = await plugin({ markdownAST });
    expect(parseASTToMarkdown(processedAST)).toMatchInlineSnapshot(`
      "# Don't Delete:

      For some reason we need at least one markdown file in the markdown-pages directory in order for the build to succeed

      <Alert header=\\"Tip\\" variant=\\"help\\">This is a standard TIP block.</Alert>
      "
    `);
  });
  it('can transform multi-line tags', async () => {
    const markdownAST = getMarkdownASTForFile('admonitions-multiline', true);
    const processedAST = await plugin({ markdownAST });
    expect(parseASTToMarkdown(processedAST)).toMatchInlineSnapshot(`
      "# Don't Delete:

      For some reason we need at least one markdown file in the markdown-pages directory in order for the build to succeed

      <Alert header=\\"Warning\\" variant=\\"warning\\">This is a standard WARNING block.
      It goes on for a few lines.
      Then we finish with some more text.</Alert>

      Like So
      "
    `);
  });
  it('can transform inline tags', async () => {
    const markdownAST = getMarkdownASTForFile('admonitions-inline', true);
    const processedAST = await plugin({ markdownAST });
    expect(parseASTToMarkdown(processedAST)).toMatchInlineSnapshot(`
      "# Don't Delete:

      For some reason we need at least one markdown file in the markdown-pages directory in order for the build to succeed

      <Alert header=\\"Note\\" variant=\\"info\\"> Only a user with sufficient rights can create a property. See <a href=\\"user-permissions.md\\">User Management</a>.</Alert>

      Like So
      "
    `);
  });
  it('can transform tags with embedded urls', async () => {
    const markdownAST = getMarkdownASTForFile('admonitions-url', true);
    const processedAST = await plugin({ markdownAST });
    expect(parseASTToMarkdown(processedAST)).toMatchInlineSnapshot(`
      "# Don't Delete:

      For some reason we need at least one markdown file in the markdown-pages directory in order for the build to succeed

      <Alert header=\\"Warning\\" variant=\\"warning\\">Before <a href=\\"https://example.com\\">website</a> After</Alert>
      "
    `);
  });
});

/*
describe("tabbedCodeBlocks", () => {
  it("is truthy", () => {
    expect(tabbedCodeBlocks).toBeTruthy();
  });
  it("can convert multiple code blocks into tabs", async () => {
    const markdownAST = getMarkdownASTForFile("tabbed-code-blocks", true);
    const processedAST = await plugin({ markdownAST });
    expect(parseASTToMarkdown(processedAST)).toMatchInlineSnapshot(`
      "# Tabbed code blocks

      Some Code

      <TabView
              orientation=\\"horizontal\\"
            >
            <Tab
                disabled={false}
                icon={undefined}
                invalid={false}
                renderChildren
                selected={false}
                tabIndex={0}
                label=\\"json\\"
              >
                {
        \\"op\\": \\"copy\\",
        \\"target\\": {
          \\"path\\": \\"/content/assets/7d391c3c-a3d4-4f66-81ce-40066732db60/target\\"
        },
        \\"source\\": {
          \\"path\\": \\"/content/assets/7d391c3c-a3d4-4f66-81ce-40066732db60/source/test.psd\\"
        }
      }
              </Tab>
              <Tab
                disabled={false}
                icon={undefined}
                invalid={false}
                renderChildren
                selected={false}
                tabIndex={0}
                label=\\"json\\"
              >
                {
        \\"op\\": \\"copy\\",
        \\"target\\": {
          \\"path\\": \\"/content/assets/7d391c3c-a3d4-4f66-81ce-40066732db60/target/test.psd\\"
        },
        \\"source\\": {
          \\"path\\": \\"/content/assets/7d391c3c-a3d4-4f66-81ce-40066732db60/source/test.psd\\"
        }
      }
              </Tab>
              </TabView>

      \`\`\`json tab-group='hello-world-snippets' tab-name='ruby'
      remove
      \`\`\`

      \`\`\`json
      {
        \\"op\\": \\"copy\\",
        \\"target\\": {
          \\"path\\": \\"/content/assets/7d391c3c-a3d4-4f66-81ce-40066732db60/target/test.psd\\"
        },
        \\"source\\": {
          \\"path\\": \\"/content/assets/7d391c3c-a3d4-4f66-81ce-40066732db60/source/test.psd\\"
        }
      }
      \`\`\`
      "
    `);
  });
});
*/
