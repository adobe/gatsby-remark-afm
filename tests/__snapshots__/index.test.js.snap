// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cleanHtml can detect and fix these open HTML tags: br, hr, and img 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 24,
              "line": 1,
              "offset": 23,
            },
            "indent": Array [],
            "start": Object {
              "column": 3,
              "line": 1,
              "offset": 2,
            },
          },
          "type": "text",
          "value": "SDS API Documentation",
        },
      ],
      "depth": 1,
      "position": Position {
        "end": Object {
          "column": 24,
          "line": 1,
          "offset": 23,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "heading",
    },
    Object {
      "position": Position {
        "end": Object {
          "column": 183,
          "line": 2,
          "offset": 206,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 24,
        },
      },
      "type": "html",
      "value": "<hr/><br/>This documentation describes SDS APIs, that can be broadly categorized into : Index, Ops, ResolveBy. For any questions please contact acp-cs-sds-engineering@adobe.com<br/><br/>",
    },
    Object {
      "position": Position {
        "end": Object {
          "column": 29,
          "line": 4,
          "offset": 236,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 4,
          "offset": 208,
        },
      },
      "type": "html",
      "value": "<img src=\\"images/image.png\\"/>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 5,
      "offset": 237,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`cleanHtml does nothing with the properly closed br, hr, and img tags in a markdown file 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 24,
              "line": 1,
              "offset": 23,
            },
            "indent": Array [],
            "start": Object {
              "column": 3,
              "line": 1,
              "offset": 2,
            },
          },
          "type": "text",
          "value": "SDS API Documentation",
        },
      ],
      "depth": 1,
      "position": Position {
        "end": Object {
          "column": 24,
          "line": 1,
          "offset": 23,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "heading",
    },
    Object {
      "position": Position {
        "end": Object {
          "column": 183,
          "line": 2,
          "offset": 206,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 24,
        },
      },
      "type": "html",
      "value": "<hr/><br/>This documentation describes SDS APIs, that can be broadly categorized into : Index, Ops, ResolveBy. For any questions please contact acp-cs-sds-engineering@adobe.com<br/><br/>",
    },
    Object {
      "position": Position {
        "end": Object {
          "column": 29,
          "line": 4,
          "offset": 236,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 4,
          "offset": 208,
        },
      },
      "type": "html",
      "value": "<img src=\\"images/image.png\\"/>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 5,
      "offset": 237,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`cleanHtml works with a mix of properly closed and open br, hr, and img tags in a markdown file 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 24,
              "line": 1,
              "offset": 23,
            },
            "indent": Array [],
            "start": Object {
              "column": 3,
              "line": 1,
              "offset": 2,
            },
          },
          "type": "text",
          "value": "SDS API Documentation",
        },
      ],
      "depth": 1,
      "position": Position {
        "end": Object {
          "column": 24,
          "line": 1,
          "offset": 23,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "heading",
    },
    Object {
      "position": Position {
        "end": Object {
          "column": 184,
          "line": 2,
          "offset": 207,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 24,
        },
      },
      "type": "html",
      "value": "<hr/><br/>This documentation describes SDS APIs, that can be broadly categorized into : Index, Ops, ResolveBy. For any questions please contact acp-cs-sds-engineering@adobe.com<br/><br/>",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 13,
              "line": 4,
              "offset": 221,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 4,
              "offset": 209,
            },
          },
          "type": "text",
          "value": "Simple text.",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 13,
          "line": 4,
          "offset": 221,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 4,
          "offset": 209,
        },
      },
      "type": "paragraph",
    },
    Object {
      "position": Position {
        "end": Object {
          "column": 46,
          "line": 6,
          "offset": 268,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 6,
          "offset": 223,
        },
      },
      "type": "html",
      "value": "<img src=\\"images/image.png\\" alt=\\"my image\\" />",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 27,
              "line": 8,
              "offset": 296,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 8,
              "offset": 270,
            },
          },
          "type": "text",
          "value": "The caption for the image.",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 27,
          "line": 8,
          "offset": 296,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 8,
          "offset": 270,
        },
      },
      "type": "paragraph",
    },
    Object {
      "position": Position {
        "end": Object {
          "column": 6,
          "line": 10,
          "offset": 303,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 10,
          "offset": 298,
        },
      },
      "type": "html",
      "value": "<hr/>",
    },
    Object {
      "position": Position {
        "end": Object {
          "column": 29,
          "line": 12,
          "offset": 333,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 12,
          "offset": 305,
        },
      },
      "type": "html",
      "value": "<img src=\\"images/image.png\\"/>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 13,
      "offset": 334,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`includeRelative can detect and render include within markdown which itself has an include 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 7,
              "line": 1,
              "offset": 6,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "text",
          "value": "before",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 7,
          "line": 1,
          "offset": 6,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 10,
              "line": 3,
              "offset": 17,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 8,
            },
          },
          "type": "text",
          "value": "more text",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 10,
          "line": 3,
          "offset": 17,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 8,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 10,
                  "line": 1,
                  "offset": 9,
                },
                "indent": Array [],
                "start": Object {
                  "column": 1,
                  "line": 1,
                  "offset": 0,
                },
              },
              "type": "text",
              "value": "Some text",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 10,
              "line": 1,
              "offset": 9,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "paragraph",
        },
        Object {
          "children": undefined,
          "position": Position {
            "end": Object {
              "column": 36,
              "line": 5,
              "offset": 61,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 11,
            },
          },
          "type": "html",
          "value": "<Alert header=\\"Caution\\" variant=\\"error\\">This is a standard CAUTION block.</Alert>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 37,
          "line": 5,
          "offset": 55,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 5,
          "offset": 19,
        },
      },
      "type": "include",
    },
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 6,
              "line": 7,
              "offset": 62,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 7,
              "offset": 57,
            },
          },
          "type": "text",
          "value": "after",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 6,
          "line": 7,
          "offset": 62,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 7,
          "offset": 57,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 8,
      "offset": 63,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`includeRelative can properly render admonitions within the contents of the included files 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 28,
                  "line": 1,
                  "offset": 27,
                },
                "indent": Array [],
                "start": Object {
                  "column": 4,
                  "line": 1,
                  "offset": 3,
                },
              },
              "type": "text",
              "value": "acppath_delete_asset_top",
            },
          ],
          "depth": 2,
          "position": Position {
            "end": Object {
              "column": 28,
              "line": 1,
              "offset": 27,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "heading",
        },
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 32,
                  "line": 3,
                  "offset": 60,
                },
                "indent": Array [],
                "start": Object {
                  "column": 1,
                  "line": 3,
                  "offset": 29,
                },
              },
              "type": "text",
              "value": "Some text for delete asset top.",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 32,
              "line": 3,
              "offset": 60,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 29,
            },
          },
          "type": "paragraph",
        },
        Object {
          "children": undefined,
          "position": Position {
            "end": Object {
              "column": 36,
              "line": 7,
              "offset": 112,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 5,
              "offset": 62,
            },
          },
          "type": "html",
          "value": "<Alert header=\\"Caution\\" variant=\\"error\\">This is a standard CAUTION block.</Alert>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 58,
          "line": 1,
          "offset": 57,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "include",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 31,
                  "line": 1,
                  "offset": 30,
                },
                "indent": Array [],
                "start": Object {
                  "column": 4,
                  "line": 1,
                  "offset": 3,
                },
              },
              "type": "text",
              "value": "acppath_delete_asset_params",
            },
          ],
          "depth": 2,
          "position": Position {
            "end": Object {
              "column": 31,
              "line": 1,
              "offset": 30,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "heading",
        },
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 28,
                  "line": 3,
                  "offset": 59,
                },
                "indent": Array [],
                "start": Object {
                  "column": 1,
                  "line": 3,
                  "offset": 32,
                },
              },
              "type": "text",
              "value": "Good text for a good topic.",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 28,
              "line": 3,
              "offset": 59,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 32,
            },
          },
          "type": "paragraph",
        },
        Object {
          "children": undefined,
          "position": Position {
            "end": Object {
              "column": 36,
              "line": 7,
              "offset": 111,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 5,
              "offset": 61,
            },
          },
          "type": "html",
          "value": "<Alert header=\\"Caution\\" variant=\\"error\\">This is a standard CAUTION block.</Alert>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 69,
          "line": 3,
          "offset": 127,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 59,
        },
      },
      "type": "include",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 31,
                  "line": 1,
                  "offset": 30,
                },
                "indent": Array [],
                "start": Object {
                  "column": 4,
                  "line": 1,
                  "offset": 3,
                },
              },
              "type": "text",
              "value": "acppath_delete_asset_bottom",
            },
          ],
          "depth": 2,
          "position": Position {
            "end": Object {
              "column": 31,
              "line": 1,
              "offset": 30,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "heading",
        },
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 41,
                  "line": 3,
                  "offset": 72,
                },
                "indent": Array [],
                "start": Object {
                  "column": 1,
                  "line": 3,
                  "offset": 32,
                },
              },
              "type": "text",
              "value": "Some text for delete asset bottom topic.",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 41,
              "line": 3,
              "offset": 72,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 32,
            },
          },
          "type": "paragraph",
        },
        Object {
          "children": undefined,
          "position": Position {
            "end": Object {
              "column": 36,
              "line": 7,
              "offset": 124,
            },
            "indent": Array [
              1,
              1,
            ],
            "start": Object {
              "column": 1,
              "line": 5,
              "offset": 74,
            },
          },
          "type": "html",
          "value": "<Alert header=\\"Caution\\" variant=\\"error\\">This is a standard CAUTION block.</Alert>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 61,
          "line": 5,
          "offset": 189,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 5,
          "offset": 129,
        },
      },
      "type": "include",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 6,
      "offset": 190,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
